internal/tiltfile/tiltfile_state.go

```diff
@@ -758,6 +799,15 @@ type k8sObjectSelector struct {
 	namespace *regexp.Regexp
 }
 
+// change a regex to only match full strings
+func reAnchor(s string) string {
+	if s == "" {
+		return s
+	} else {
+		return fmt.Sprintf("^(?:%s)$", s)
```

Comment:
this seems very brittle? like if the user made a typo in the regexp, or forgot a paren, the error message would be busted